### PR TITLE
Remove gcloud.dev usage from Google Cloud Storage

### PR DESCRIPTION
**Pull request by lavoiesl**
`gcloud.dev` brings a _ton_ of dependencies that is not necessarily useful for this project.

On top of it `gcloud.dev` has a bunch of logic that can further complicate troubleshooting

This changes `cloudStorageFS` to use `cloud.google.com/go/storage`, paving the way to removing the dependency altogether.

This is in preparation to bumping go-storage to v2

### CODE REVIEW

File: cloudstorage_fs.go [hunk 1]
```diff
@@ -162,86 +180,80 @@ func (c *cloudStorageFS) Walk(ctx context.Context, path string, fn WalkFn) error
...
 		return nil, errors.Wrap(err, "cloud storage: parse credentials")
 	}
 
-	return &gcsblob.Options{
-		PrivateKey:     config.PrivateKey,
-		GoogleAccessID: config.Email,
-	}, nil
+	// Atomically store, it's fine if multiple threads end up storing at the same time, they are all equivalent
+	ptr := (*unsafe.Pointer)(unsafe.Pointer(&c.jwtConfig))
+	atomic.StorePointer(ptr, unsafe.Pointer(config))
```
[the remark this answers is not in the document]
Comment (lavoiesl) [reply cloudstorage_fs.go]: I ended up preferring atomic 👍

[the remark this answers is not in the document]
Comment (lavoiesl) [reply cloudstorage_fs.go]: The implementation now uses a newer version of the storage api, which exposes https://pkg.go.dev/cloud.google.com/go/storage#BucketHandle.SignedURL, so this code path is now completely removed.

**Review by hurracrane — COMMENTED** [review]
As a first pass, I am noticing this PR is trying to adapt the new library so that it behaves the same. It looks like code is mapped correctly. Is there anything else I should be looking out for?

**Comment by lavoiesl** [comment]
That's the idea, yes. I'm not aware of any gotchas that should be carefully considered; it's simply that it's a significant number of changes.

**Comment by yousifh** [comment]
I think this looks fine? I tried to compare it to the older implementation with gocloud.dev but I haven't dived too deep into it. Creating the client does seem much simpler now

**Review by yousifh — APPROVED** [review]

**Review by pior — APPROVED** [review]
👍

**Pull request merged**